data_warehouse/views.py

```diff
@@ -28,3 +30,20 @@ def get_canvas_options(self, request):
         '''getting canvas-options'''
         data = CanvasOptionsService.get_canvas_options()
         return Response(data)
+
+    @action(detail=False, methods=['GET'], url_path='table-export',
+            url_name='table-export')
+    def export(self, request):
+        '''Return a xls file stream.'''
+        if 'table_type' not in request.GET:
+            raise BadRequest('请求的参数不正确。')
+        table_type = request.GET['table_type']
+        table_type = int(table_type)
+        context = {key: val for (key, val) in request.GET.items()}
+        context['request'] = request
+        ret_file_path = AggregateDataService.dispatch('table_export', context)
+        file_name = AggregateDataService.TABLE_NAME_CHOICES[
+            table_type] + '.xls'
+        secure_file = SecureFile.from_path(request.user, file_name,
+                                           ret_file_path)
```

Comment:
According to the doc.
> Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it.

You should call `os.unlink(path)` to remove the file after SecureFile is created.